### PR TITLE
Ens parameters

### PR DESCRIPTION
**Pull request by Siegrift**
Fixes https://api3workspace.slack.com/archives/C02ALA11APM/p1628590492006400 

which was requested https://forum.api3.org/t/ens-name-for-proposal-creator/289

Fixes https://github.com/api3dao/api3-dao-dashboard/issues/224
Fixes https://github.com/api3dao/api3-dao-dashboard/issues/223

### CODE REVIEW

**Comment by Siegrift** [comment]
I am so happy that I wrote the e2e test because I introduced a hard to spot bug while implementing this.
Check out last commit to see it.

File: src/logic/proposals/encoding.ts [hunk 1]
```diff
@@ -170,14 +180,21 @@ export const decodeEvmScript = (script: string, metadata: ProposalMetadata): Dec
...
+        return ensName || param;
+      })
+    );
 
     return {
       targetAddress: targetContractAddress,
       value,
-      rawParameters,
-      parameters: stringifyBigNumbersRecursively(rawParameters),
+      parameters: stringifyBigNumbersRecursively([...parameters]),
```
Comment (andreogle) [inline src/logic/proposals/encoding.ts:197]: The "destructuring to enforce Array shape" comment got lost. Is it still necessary to spread here vs just having `stringifyBigNumbersRecursively(parameters)`?

File: src/pages/proposals/forms/new-proposal-form.tsx [hunk 2]
```diff
@@ -194,7 +196,8 @@ const NewProposalForm = (props: Props) => {
               title,
             };
 
-            if (!validateForm(formData)) {
+            const containsError = await validateForm(formData);
+            if (!containsError) {
```
Comment (andreogle) [inline src/pages/proposals/forms/new-proposal-form.tsx:200]: Nice catch with the subtle bug

**Review by andreogle — DISMISSED** [review]

File: src/logic/proposals/encoding.ts (hunk 1 above)
Comment (Siegrift) [reply src/logic/proposals/encoding.ts:197]: I don't think we need that.

**Review by andreogle — APPROVED** [review]

**Pull request merged**